docs/user/08-10-best-practices.md

```diff
@@ -0,0 +1,61 @@
+Defining a Function in Kyma essentially means creating a new instance of the Function custom resource (CR). However, the content of the Function CR specification may become quite long. It consists of the code (or git reference to the code), dependencies, runtime specification, build-time specification, etc. Additionally, there are other CRs that are relevant for a Function developer - i.e. [APIRule](https://kyma-project.io/docs/kyma/latest/05-technical-reference/00-custom-resources/apix-01-apirule/) (defining how Function is exposed to the outside world), [Subscription](https://kyma-project.io/docs/kyma/latest/05-technical-reference/00-custom-resources/evnt-01-subscription/) (defining which CloudEvents should trigger Function) and others.
+
+
+All of that can be easily managed using the following best practices for the Function development. You will find hints that will be helpful for you at any stage of your development journey.
+
+# Use UI to explore
+
+At the beginning of your Kyma journey, you will probably want to evaluate Serverless and draft a few Functions.
+Kyma Dashboard is perfect to gain basic experience and start the journey with Kyma Functions. The Dashboard consists of UI components dedicated to Serverless. They will help you draft your first Functions by putting the code directly in the browser using a Web IDE.
+Kyma Dashboard will also help you expose your Function using HTTP, define environment variables, subscribe to CloudEvents, bind ServiceInstances, and even show you the Function logs - all in one place.
+
+Get started with [Function UI](/docs/user/01-10-create-inline-function.md)
+![function-ui](../assets/svls-function-ui.png)
+
+# Use Kyma CLI for better development experience
+
+Defining your Function from the UI is very quick and easy, but it might not be enough to satisfy your needs as a developer. To code and test more complex cases, you may want to write your Function in your favorite IDE or run and debug the Function on your local machine, before actually deploying in Kyma runtime. Also, you would probably want to avoid recreating the same Functions manually from the UI on a different environment. In the end, having deployable artifacts is more desirable. This is where Kyma CLI comes in handy, as it allows you to keep your Function's code and configuration in the form of a workspace. 
+
+Initialize a scaffold for a brand new Function using the `kyma init function` command or fetch the current state of an existing Function deployed in your Kyma runtime using `kyma sync function`.
+Focus on the Function code and develop it from your favorite IDE. Configure your Functions directly in the [`config.yaml` manifest file](/docs/user/07-60-function-configuration-file.md)
+>**TIP:** Use `kyma init function --vscode` to generate json schema which can be used in VSCode for autocompletion.
```

Comment:
```suggestion
>**TIP:** Use `kyma init function --vscode` to generate a `.json` schema, which can be used in VSCode for autocompletion.
```

[code font for file extensions](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md)